pkg/waiter/manager.go

```diff
@@ -0,0 +1,255 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const (
+	// The default number of attempts to re-generate an ID on collision.
+	defaultMaxCollisions = 5
+)
+
+var (
+	// ErrTooManyCollisions the ID generator produced too many non-unique IDs.
+	ErrTooManyCollisions = errors.New("too many id collisions")
+
+	// ErrManagerShutdown manager has been shutdown and therefore cannot process requests.
+	ErrManagerShutdown = errors.New("manager is shutdown")
+
+	errKeyExists = errors.New("id exists")
+	errNoExists  = errors.New("id does not exist")
```

Comment:
nit: could align this to `errox.NotFound`.

Reply:
Both these errors have been removed in the pending push, but will note for next time!